File: frontend/apps/ios/Pantopus/Features/Place/Detail/PlaceDetailPrimitives.swift
//
//  PlaceDetailPrimitives.swift
//  Pantopus
//
//  Shared chrome for the Place group-detail pages — ported from the
//  design kit's ContentDetail layout (place-*-detail.jsx): the sticky
//  blurred header (back chevron + title + address), the section-label
//  overline, the provider Source caption, and small fact primitives the
//  bespoke detail cards compose.
//

import SwiftUI

// MARK: - Detail header (back + title + address)

struct PlaceDetailHeader: View {
    let title: String
    let address: String
    var onBack: () -> Void

    var body: some View {
        HStack(spacing: 10) {
            Button(action: onBack) {
                Icon(.chevronLeft, size: 20, strokeWidth: 2.5, color: Theme.Color.appTextStrong)
                    .frame(width: 34, height: 34)
                    .background(Theme.Color.appSurface)
                    .clipShape(Circle())
                    .shadow(color: .black.opacity(0.06), radius: 1, x: 0, y: 1)
            }
            .buttonStyle(.plain)
            .accessibilityLabel("Back")
            VStack(alignment: .leading, spacing: 1) {
                Text(title)
                    .font(.system(size: 20, weight: .bold))
                    .kerning(-0.4)
                    .foregroundStyle(Theme.Color.appText)
                Text(address)
                    .font(.system(size: 12.5, weight: .medium))
                    .foregroundStyle(Theme.Color.appTextMuted)
                    .lineLimit(1)
            }
            Spacer(minLength: 0)
        }
        .padding(.horizontal, 14)
        .padding(.top, 6)
        .padding(.bottom, 12)
        .background(.bar)
        .overlay(alignment: .bottom) {
            Rectangle().fill(Theme.Color.appBorder).frame(height: 1)
        }
    }
}

/// Compact header address: "1421 SE Oak St · Portland".
func placeDetailAddress(_ place: PlaceAddressRef) -> String {
    let line1 = place.line1.trimmingCharacters(in: .whitespaces)
    let city = place.city.trimmingCharacters(in: .whitespaces)
    if !line1.isEmpty, !city.isEmpty { return "\(line1) · \(city)" }
    return line1.isEmpty ? place.label : line1
}

// MARK: - Section overline

struct PlaceDetailSectionLabel: View {
    let text: String

    var body: some View {
        Text(text.uppercased())
            .font(.system(size: 11, weight: .bold))
            .kerning(0.88)
            .foregroundStyle(Theme.Color.appTextMuted)
            .padding(.horizontal, 4)
            .padding(.top, 26)
            .padding(.bottom, 9)
            .frame(maxWidth: .infinity, alignment: .leading)
            .accessibilityAddTraits(.isHeader)
    }
}

// MARK: - Provider source caption

struct PlaceSourceNote: View {
    let name: String
    var asOf: String?

    var body: some View {
        HStack(spacing: 6) {
            Text(name)
                .fontWeight(.medium)
            if let asOf {
                Text("·").opacity(0.5)
                Text(asOf)
            }
        }
        .font(Theme.Font.caption)
        .foregroundStyle(Theme.Color.appTextMuted)
        .padding(.top, 10)
        .frame(maxWidth: .infinity, alignment: .leading)
    }
}

// MARK: - Bare card surface (detail cards compose into it)

struct PlaceDetailCard<Content: View>: View {
    var padding: CGFloat = 18
    @ViewBuilder var content: Content

    var body: some View {
        content
            .frame(maxWidth: .infinity, alignment: .leading)
            .padding(padding)
            .placeCard()
    }
}

// MARK: - A labelled fact (2×2 grid cell)

struct PlaceFactCell: View {
    let icon: PantopusIcon
    let label: String
    let value: String

    var body: some View {
        HStack(spacing: 9) {
            PlaceIconTile(icon: icon, tone: .home, size: 30)
            VStack(alignment: .leading, spacing: 1) {
                Text(label)
                    .pantopusTextStyle(.caption)
                    .foregroundStyle(Theme.Color.appTextMuted)
                Text(value)
                    .font(.system(size: 15, weight: .semibold))
                    .foregroundStyle(Theme.Color.appText)
                    .lineLimit(1)
            }
            Spacer(minLength: 0)
        }
    }
}

// MARK: - "Coming soon" placeholder row (uncovered/BUILD_PENDING)

struct PlaceComingSoonRow: View {
    let icon: PantopusIcon
    let title: String
    var subtitle: String?

    var body: some View {
        HStack(spacing: 11) {
            PlaceIconTile(icon: icon, tone: .muted, size: 32)
            VStack(alignment: .leading, spacing: 1) {
                Text(title)
                    .font(.system(size: 14.5, weight: .semibold))
                    .foregroundStyle(Theme.Color.appTextSecondary)
                if let subtitle {
                    Text(subtitle)
                        .font(.system(size: 12.5))
                        .foregroundStyle(Theme.Color.appTextMuted)
                }
            }
            Spacer(minLength: 0)
            PlaceChip(model: PlaceChipModel(tone: .neutral, text: "Coming soon"))
        }
        .padding(14)
        .placeCard()
    }
}

File: frontend/apps/android/app/src/main/java/app/pantopus/android/ui/screens/place/detail/PlaceDetailPrimitives.kt
package app.pantopus.android.ui.screens.place.detail

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.semantics.heading
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import app.pantopus.android.data.api.models.place.PlaceAddressRef
import app.pantopus.android.ui.screens.place.components.PlaceChip
import app.pantopus.android.ui.screens.place.components.PlaceChipModel
import app.pantopus.android.ui.screens.place.components.PlaceChipTone
import app.pantopus.android.ui.screens.place.components.PlaceIconTile
import app.pantopus.android.ui.screens.place.components.PlaceTileTone
import app.pantopus.android.ui.screens.place.components.placeCard
import app.pantopus.android.ui.theme.PantopusColors
import app.pantopus.android.ui.theme.PantopusIcon
import app.pantopus.android.ui.theme.PantopusIconImage

/**
 * Shared chrome for the Place group-detail pages — parity twin of iOS
 * `PlaceDetailPrimitives`: the header (back + title + address), the
 * section overline, the provider source caption, and small fact rows.
 */

@Composable
fun PlaceDetailHeader(
    title: String,
    address: String,
    onBack: () -> Unit,
) {
    Row(
        modifier =
            Modifier
                .fillMaxWidth()
                .padding(horizontal = 14.dp)
                .padding(top = 6.dp, bottom = 12.dp),
        horizontalArrangement = Arrangement.spacedBy(10.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Box(
            modifier =
                Modifier
                    .size(34.dp)
                    .shadow(1.dp, CircleShape, clip = false)
                    .clip(CircleShape)
                    .background(PantopusColors.appSurface)
                    .clickable(onClick = onBack),
            contentAlignment = Alignment.Center,
        ) {
            PantopusIconImage(
                icon = PantopusIcon.ChevronLeft,
                contentDescription = "Back",
                size = 20.dp,
                strokeWidth = 2.5f,
                tint = PantopusColors.appTextStrong,
            )
        }
        Column(modifier = Modifier.weight(1f)) {
            Text(
                text = title,
                fontSize = 20.sp,
                fontWeight = FontWeight.Bold,
                letterSpacing = (-0.4).sp,
                lineHeight = 24.sp,
                color = PantopusColors.appText,
            )
            Text(
                text = address,
                fontSize = 12.5.sp,
                fontWeight = FontWeight.Medium,
                color = PantopusColors.appTextMuted,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
            )
        }
    }
}

fun placeDetailAddress(place: PlaceAddressRef): String {
    val line1 = place.line1.trim()
    val city = place.city.trim()
    return if (line1.isNotEmpty() && city.isNotEmpty()) "$line1 · $city" else line1.ifEmpty { place.label }
}

@Composable
fun PlaceDetailSectionLabel(text: String) {
    Text(
        text = text.uppercase(),
        fontSize = 11.sp,
        fontWeight = FontWeight.Bold,
        letterSpacing = 0.88.sp,
        color = PantopusColors.appTextMuted,
        modifier =
            Modifier
                .fillMaxWidth()
                .padding(horizontal = 4.dp)
                .padding(top = 26.dp, bottom = 9.dp)
                .semantics { heading() },
    )
}

@Composable
fun PlaceSourceNote(
    name: String,
    asOf: String? = null,
) {
    Row(
        modifier = Modifier.fillMaxWidth().padding(top = 10.dp),
        horizontalArrangement = Arrangement.spacedBy(6.dp),
        verticalAlignment = Alignment.CenterVertically,
    ) {
        Text(text = name, fontSize = 12.sp, fontWeight = FontWeight.Medium, color = PantopusColors.appTextMuted)
        if (asOf != null) {
            Text("·", fontSize = 12.sp, color = PantopusColors.appTextMuted.copy(alpha = 0.5f))
            Text(asOf, fontSize = 12.sp, color = PantopusColors.appTextMuted)
        }
    }
}

@Composable
fun PlaceDetailCard(
    modifier: Modifier = Modifier,
    padding: Dp = 18.dp,
    content: @Composable androidx.compose.foundation.layout.ColumnScope.() -> Unit,
) {
    Column(
        modifier = modifier.fillMaxWidth().placeCard().padding(padding),
        content = content,
    )
}

@Composable
fun PlaceFactCell(
    icon: PantopusIcon,
    label: String,
    value: String,
    modifier: Modifier = Modifier,
) {
    Row(modifier = modifier, horizontalArrangement = Arrangement.spacedBy(9.dp), verticalAlignment = Alignment.CenterVertically) {
        PlaceIconTile(icon = icon, tone = PlaceTileTone.HOME, size = 30.dp)
        Column {
            Text(text = label, fontSize = 12.sp, color = PantopusColors.appTextMuted)
            Text(
                text = value,
                fontSize = 15.sp,
                fontWeight = FontWeight.SemiBold,
                color = PantopusColors.appText,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
            )
        }
    }
}

@Composable
fun PlaceComingSoonRow(
    icon: PantopusIcon,
    title: String,
    subtitle: String? = null,
) {
    PlaceDetailCard(padding = 14.dp) {
        Row(horizontalArrangement = Arrangement.spacedBy(11.dp), verticalAlignment = Alignment.CenterVertically) {
            PlaceIconTile(icon = icon, tone = PlaceTileTone.MUTED, size = 32.dp)
            Column(modifier = Modifier.weight(1f)) {
                Text(text = title, fontSize = 14.5.sp, fontWeight = FontWeight.SemiBold, color = PantopusColors.appTextSecondary)
                if (subtitle != null) {
                    Text(text = subtitle, fontSize = 12.5.sp, color = PantopusColors.appTextMuted)
                }
            }
            PlaceChip(PlaceChipModel(PlaceChipTone.NEUTRAL, "Coming soon"))
        }
    }
}

/** A small status dot used in stat rows. */
@Composable
fun PlaceDot(
    color: Color,
    size: Dp = 6.dp,
) {
    Box(modifier = Modifier.size(size).clip(CircleShape).background(color))
}
